File: BackEnd.Net/src/Alcohol/Models/Discount.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Alcohol.Models;

public class Discount
{
    [Key]
    public int Id { get; set; }

    [Required]
    [StringLength(50)]
    public string Code { get; set; }

    public string Description { get; set; }

    [Required]
    public decimal DiscountAmount { get; set; }

    [Required]
    public decimal MinimumOrderAmount { get; set; }

    [Required]
    public DateTime StartDate { get; set; }

    [Required]
    public DateTime EndDate { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    public ICollection<Product> Products { get; set; }

    public Discount()
    {
        Products = new List<Product>();
    }
} 

File: FrontEndNext/frontendnext/src/lib/types/discount.ts
export interface Discount {
  id: number;
  code: string;
  name: string;
  description: string;
  discountType: number; // 0: Percentage, 1: Fixed Amount
  discountValue: number;
  minimumOrderAmount: number;
  maximumDiscountAmount: number;
  startDate: string;
  endDate: string;
  usageLimit: number;
  usedCount: number;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface DiscountCreateDto {
  code: string;
  name: string;
  description: string;
  discountType: number;
  discountValue: number;
  minimumOrderAmount: number;
  maximumDiscountAmount: number;
  startDate: string;
  endDate: string;
  usageLimit: number;
  isActive: boolean;
}

export interface DiscountUpdateDto {
  code: string;
  name: string;
  description: string;
  discountType: number;
  discountValue: number;
  minimumOrderAmount: number;
  maximumDiscountAmount: number;
  startDate: string;
  endDate: string;
  usageLimit: number;
  isActive: boolean;
}
